### PR TITLE
Add tests for incremental preconditions

### PR DESCRIPTION
**Pull request by cdmihai**
And fix up incremental preconditions issues revealed by tests

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2725)

### CODE REVIEW

**Comment by cdmihai** [comment]
@piotrpMSFT

**Comment by cdmihai** [comment]
Disable the muxer resolver since it seemed to always resolve and thus skip the other resolvers. Also, it returned a resolution strategy of RootedPath. But I see that non windows tests now fail :(

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/CommandResultAssertions.cs [hunk 1]
```diff
@@ -114,14 +114,14 @@ private string AppendDiagnosticsTo(string s)
...
-            _commandResult.StdOut.Should().Contain($"Project {skippedProject} ({frameworkFullName}) was previously compiled. Skipping compilation.");
+            _commandResult.StdOut.Contains($"Project {skippedProject} ({frameworkFullName}) was previously compiled. Skipping compilation.");
 
             return new AndConstraint<CommandResultAssertions>(this);
         }
 
         public AndConstraint<CommandResultAssertions> HaveCompiledProject(string compiledProject, string frameworkFullName)
         {
-            _commandResult.StdOut.Should().Contain($"Project {compiledProject} ({frameworkFullName}) will be compiled");
+            _commandResult.StdOut.Contains($"Project {compiledProject} ({frameworkFullName}) will be compiled");
```
Comment (pakrym) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/CommandResultAssertions.cs]: Isn't this `string.Contains`? If yes, you are not using it's result.

Comment (cdmihai) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Assertions/CommandResultAssertions.cs]: Whoops. :)

**Comment by eerhardt** [comment]
@cdmihai - are you still interested in making this change?  If so, please rebase and get green.

I'll close this out if I don't hear back by tomorrow.

**Comment by cdmihai** [comment]
I don't think this is relevant anymore. Closing the PR.

**Pull request closed without merging**